bapsflib/_hdf/maps/controls/tests/fauxnixyz.py

```diff
@@ -0,0 +1,311 @@
+            # return self._faux.n_configs
+            return self._n_configs
+
+        @property
+        def n_motionlists(self):
+            """Number of motion lists"""
+            return self._n_motionlists
+
+        @n_motionlists.setter
+        def n_motionlists(self, val: int):
+            """Set number of motion lists"""
+            # check if self._n_motionlists has been defined
+            if hasattr(self, '_n_motionlists'):
+                old_val = self._n_motionlists
+            else:
+                old_val = 1
+
+            # condition val
+            if not isinstance(val, (int, np.integer)):
+                warn("Setting value must be an integer of >=1")
+                val = old_val
+            elif val < 1:
+                warn("Setting value must be an integer of >=1")
+                val = old_val
+
+            # only update if self._n_motionlists had been defined once
+            # prior
+            if hasattr(self, '_n_motionlists'):
+                self._n_motionlists = val
+                self._faux.populate()
+            else:
+                self._n_motionlists = val
+
+        @property
+        def n_probes(self):
+            """Number of controlled probes"""
+            # return self._faux.n_probes
+            return self._n_probes
+
+        @property
+        def sn_size(self):
+            """Shot number size"""
+            return self._sn_size
+
+        @sn_size.setter
+        def sn_size(self, val: int):
+            """Set shot number size"""
+            # check if self._sn_size has been defined
+            if hasattr(self, '_sn_size'):
+                old_val = self._sn_size
+            else:
+                old_val = 100
+
+            # condition val
+            if not isinstance(val, (int, np.integer)):
+                warn("Setting value must be an integer of >=1")
+                val = old_val
+            elif val < 1:
+                warn("Setting value must be an integer of >=1")
+                val = old_val
+
+            # only update if self._sn_size had been defined once prior
+            if hasattr(self, '_sn_size'):
+                self._sn_size = val
+                self._faux.populate()
+            else:
+                self._sn_size = val
+
+        def reset(self):
+            """Reset to defaults"""
+            self._n_configs = 1
+            self._n_probes = 1
+            self._n_motionlists = 1
+            self._sn_size = 100
+            self._faux.populate()
+
+    def __init__(self, id, n_motionlists=1, sn_size=100, **kwargs):
+        # ensure id is for a HDF5 group
+        if not isinstance(id, h5py.h5g.GroupID):
+            raise ValueError('{} is not a GroupID'.format(id))
+
+        # create control group
+        gid = h5py.h5g.create(id, b'NI_XYZ')
+        h5py.Group.__init__(self, gid)
+
+        # store number on configurations
+        self._knobs = self.Knobs(self, n_motionlists, sn_size)
+
```

Comment:
improve initialization and clearing of attributes (1 of 3)
```suggestion
        # initialize some attributes
        self._ml = []  # list of motion lists
        self.configs = {}  # configurations dictionary
```